lib/node-token-validation/src/config/OpenIdConfigProvider.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { INetworkModule, Logger, NetworkResponse, ProtocolMode } from "@azure/msal-common";
+import { ValidationConfigurationError } from "../error/ValidationConfigurationError";
+import { OpenIdConfigResponse } from "../response/OpenIdConfigResponse";
+import { TokenValidationConfiguration } from "./Configuration";
+
+export class OpenIdConfigProvider {
+    protected configuration: TokenValidationConfiguration;
+    protected networkInterface: INetworkModule;
+    protected logger: Logger;
+
+    constructor(configuration: TokenValidationConfiguration, networkInterface: INetworkModule, logger: Logger) {
+        this.configuration = configuration;
+        this.networkInterface = networkInterface;
+        this.logger = logger;
+    }
+
+    async fetchJwksUriFromEndpoint(): Promise<string> {
+        this.logger.trace("OpenIdConfigProvider.fetchJwksUriFromEndpoint called");
+
+        const endpointMetadata = await this.getMetadata();
+        return endpointMetadata.body.jwks_uri;
+    }
+
+    async getMetadata(): Promise<NetworkResponse<OpenIdConfigResponse>> {
+        this.logger.trace("OpenIdConfigProvider.getMetadata called");
+
+        const endpoint = await this.getOpenIdConfigurationEndpoint(this.configuration.auth.authority, this.configuration.auth.protocolMode);
+        this.logger.verbose(`OpenIdConfigProvider - openIdConfigurationEndpoint: ${endpoint}`);
+        const endpointMetadata = await this.networkInterface.sendGetRequestAsync<OpenIdConfigResponse>(endpoint);
```

Comment:
Should we use the `Authority` class exported by msal-common here instead, since it already implements this logic?

Reply:
I explored using the `Authority` class from msal-common, but decided against it for the time being because it requires a `cacheManager` implementation, along with other `AuthorityOptions` which may not be relevant to this library. In the interest of keeping the first version of this library lightweight, I have not yet implemented caching, but may do so to address metadata refresh concerns later on, at which time it may make sense to use the `Authority` class instead.